### PR TITLE
Zamant2

### PR DESCRIPTION
**Pull request by zamant2**
Few commented lines, which I might need to uncomment later.

### CODE REVIEW

File: src/main/resources/conf/env/stage.xml
Comment (matakevin) [inline src/main/resources/conf/env/stage.xml:1]: What is the addition of the application tag on line 6 for?

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppDCEGProgressionTest.java
[the remark this answers is not in the document]
Comment (matakevin) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunAppDCEGProgressionTest.java:1]: Regression runner being tagged "tz"

Comment (georgeb79) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppDCEGProgressionTest.java:1]: Line 14 - This should have @Progression tag

File: src/test/java/CustomBusiness/EIDP/Pages/CoPrimaryMentorPage.java
Comment (matakevin) [inline src/test/java/CustomBusiness/EIDP/Pages/CoPrimaryMentorPage.java:1]: Commented sleep on line 186

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPRegressionTest.java
Comment (georgeb79) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPRegressionTest.java:1]: Line 13 - This should have @Regression tag

File: src/main/resources/conf/env/stage.xml
Comment (zamant2) [reply src/main/resources/conf/env/stage.xml:1]: Updated

File: src/test/java/CustomBusiness/EIDP/Steps/CoPrimaryMentorSteps.java
Comment (matakevin) [inline src/test/java/CustomBusiness/EIDP/Steps/CoPrimaryMentorSteps.java:1]: Commented code on like 46

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPSanityTest.java
Comment (georgeb79) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPSanityTest.java:1]: If this Runner uses a @Smoke tag I believe Smoke should be added un the file's name to be easily identifiable.

File: src/test/java/CustomBusiness/EIDP/Steps/DashboardSteps.java
Comment (matakevin) [inline src/test/java/CustomBusiness/EIDP/Steps/DashboardSteps.java:1]: Is the commented code on line 194 needed?

File: src/test/java/CustomBusiness/EIDP/Steps/IDPAwaitingResponseSteps.java
Comment (matakevin) [inline src/test/java/CustomBusiness/EIDP/Steps/IDPAwaitingResponseSteps.java:1]: Commented line 35 doesn't seem needed

File: src/test/java/CustomBusiness/EIDP/Pages/CareerGoalAndActivePage.java
Comment (georgeb79) [inline src/test/java/CustomBusiness/EIDP/Pages/CareerGoalAndActivePage.java:1]: extra space lines 11 and 42. Please remove.

File: src/test/java/CustomBusiness/EIDP/Pages/CoPrimaryMentorPage.java
Comment (georgeb79) [inline src/test/java/CustomBusiness/EIDP/Pages/CoPrimaryMentorPage.java:1]: extra space line 173 Please remove.

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppDCEGProgressionTest.java
[the remark this answers is not in the document]
Comment (zamant2) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunAppDCEGProgressionTest.java:1]: Updated

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPRegressionTest.java
[the remark this answers is not in the document]
Comment (zamant2) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPRegressionTest.java:1]: Fixed

File: src/test/java/CustomBusiness/EIDP/Pages/CoPrimaryMentorPage.java
Comment (zamant2) [reply src/test/java/CustomBusiness/EIDP/Pages/CoPrimaryMentorPage.java:1]: Updated

**Review by georgeb79 — CHANGES_REQUESTED** [review]
I left a few comments for items that need attention.

File: src/test/java/CustomBusiness/EIDP/StepsImplementation/TraineeReviewStepsImpl.java
Comment (matakevin) [inline src/test/java/CustomBusiness/EIDP/StepsImplementation/TraineeReviewStepsImpl.java:1]: Is the try catch block needed here?

**Review by matakevin — COMMENTED** [review]
Clarification is needed for the try catch block being necessary but aside from that and the hardcoded credentials actually working or not, all the rest of the code looks good

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppDCEGProgressionTest.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+@RunWith(Cucumber.class)
+@CucumberOptions(plugin= {"html:target/html-reports/cucumber-default-report"
+		, "json:target/cucumber.json"
+		, "junit:target/cucumber.xml"
+		, "rerun:target/failed.txt","pretty"}
+		, features="src/test/java/CustomBusiness/DCEG/Features"
+		, glue="CustomBusiness.DCEG.Steps"
+		, tags="@tz"
+        , dryRun =false
+        ,junit = "--step-notifications"
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppDCEGProgressionTest.java]: missing monochrome and strict

**Review by hamidnesar — CHANGES_REQUESTED** [review]
Left a few comments. I noticed Thread.sleep and MiscUtils are used interchangeably across pages. I would recommend MiscUtils.sleep for consistency

File: src/main/resources/conf/env/stage.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 	<!-- Below application ID's are for STAGE environments -->
 
 	<profile-name>automation</profile-name>
-
+	<application>
```
[the remark this answers is not in the document]
Comment (zamant2) [reply src/main/resources/conf/env/stage.xml]: updated

File: src/test/java/CustomBusiness/EIDP/Pages/CoPrimaryMentorPage.java [hunk 3]
```diff
@@ -53,6 +53,12 @@ public class CoPrimaryMentorPage extends CommonUtils {
 	@FindBy(id = "returnToPMComments")
 	private WebElement primaryMentorComments;
```
[the remark this answers is not in the document]
Comment (zamant2) [reply src/test/java/CustomBusiness/EIDP/Pages/CoPrimaryMentorPage.java:54]: There a lot of element using the private which need to be updated gradually

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPSanityTest.java
Comment (zamant2) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPSanityTest.java:1]: Updated the runner

File: src/test/java/CustomBusiness/EIDP/StepsImplementation/ProjectRelatedDeliverableStepImpl.java
[the remark this answers is not in the document]
Comment (zamant2) [reply src/test/java/CustomBusiness/EIDP/StepsImplementation/ProjectRelatedDeliverableStepImpl.java:1]: Hardcoding issue will resolve gradually as I progress through test cases

File: src/test/java/CustomBusiness/EIDP/StepsImplementation/TraineeReviewStepsImpl.java
Comment (zamant2) [reply src/test/java/CustomBusiness/EIDP/StepsImplementation/TraineeReviewStepsImpl.java:1]: removed the try catch block

File: src/test/java/CustomBusiness/ETD/StepsImplementation/ETDFlowStepsImpl.java
[the remark this answers is not in the document]
Comment (zamant2) [reply src/test/java/CustomBusiness/ETD/StepsImplementation/ETDFlowStepsImpl.java:1]: yes

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPSmokeTest.java
Comment (matakevin) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPSmokeTest.java:1]: monochrome and strict tags are missing and junit was added as a tag

Comment (matakevin) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPSmokeTest.java:1]: Tag is also "@Smoke " with a space added after

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppDCEGProgressionTest.java (hunk 1 above)
Comment (hamidnesar) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppDCEGProgressionTest.java]: missing monochrome and strict tags

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPSanityTest.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+@RunWith(Cucumber.class)
+@CucumberOptions(plugin= {"html:target/html-reports/cucumber-default-report"
+		, "json:target/cucumber.json"
+		, "junit:target/cucumber.xml"
+		, "rerun:target/failed.txt","pretty"}
+		, features="src/test/java/CustomBusiness/EIDP/Features"
+		, glue="CustomBusiness.EIDP.Steps"
+		, tags="@Smoke "
+		, dryRun = false,
+		junit = "--step-notifications")
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPSanityTest.java]: missing monochrome and strict tags

File: src/main/resources/conf/env/stage.xml
Comment (matakevin) [inline src/main/resources/conf/env/stage.xml:1]: Line 196 "<!-- add test..." should be <!-- add stage..."

**Review by matakevin — APPROVED** [review]
LGTM

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPProgressionTest.java [hunk 5]
```diff
@@ -6,9 +6,18 @@
...
+		, "junit:target/cucumber.xml"
+		, "rerun:target/failed.txt"
+		, "pretty"
+		, "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"}
+		, features = "src/test/java/CustomBusiness/EIDP/Features"
+		, glue = "CustomBusiness.EIDP.Steps"
+		, tags = "@Progression"
+		, dryRun = false
+		, monochrome = true
+		, strict = true
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppEIDPProgressionTest.java:21]: Please remove the extra line

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppETDProgressionTest.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+		, "junit:target/cucumber.xml"
+		, "rerun:target/failed.txt","pretty"}
+		, features="src/test/java/CustomBusiness/ETD/Features"
+		, glue="CustomBusiness.ETD.Steps"
+		, tags = "@Progression"
+		, dryRun = false
+		, monochrome = true
+		, strict = true		
+		)
+
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppETDProgressionTest.java:19]: Please remove the extra line

File: src/test/java/CustomBusiness/BusinessAppRunners/RunAppETDRegressionTest.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+		, "junit:target/cucumber.xml"
+		, "rerun:target/failed.txt","pretty"}
+		, features="src/test/java/CustomBusiness/ETD/Features"
+		, glue="CustomBusiness.ETD.Steps"
+		, tags = "@Regression"
+		, dryRun = false
+		, monochrome = true
+		, strict = true		
+		)
+
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunAppETDRegressionTest.java:19]: Please remove the extra line. You have closing bracket on the same line as Strict in runners while on a new line a few

**Review by hamidnesar — APPROVED** [review]
It would be nice to remove the extra lines and have the closing brackets consistent across different runner classes.

**Review by georgeb79 — APPROVED** [review]

**Review by alenaminsk — APPROVED** [review]

**Review by alenaminsk — APPROVED** [review]

**Review by alenaminsk — APPROVED** [review]

**Review by iamdez99 — APPROVED** [review]
Reviewed and approved - juarezds

**Review by mzuhoor — COMMENTED** [review]
Done

**Pull request merged**